common/src/main/java/org/iris_events/annotations/Message.java

```diff
@@ -69,4 +70,18 @@
      * they reach the queue.
      */
     boolean persistent() default false;
+
+    /**
+     * Defines whether the message acts as a RPC call request. If this is true the message should have a counterpart response
+     * message
+     * defined in the same service. The {@link #response() response} property must be set to that counterpart response message
+     * class and
+     * the {@link #rpcReplyTo() rpcReplyTo} property must be set to the name of the {@link #response() response message}
+     */
+    boolean isRpcRequest() default false;
+
+    /**
+     * Defines a reply to property of the RPC messages. Ignored if the message is not defined as RPC.
+     */
+    String rpcReplyTo() default "";
```

Comment:
rename to responseEventName

Reply:
if anything, this should be reference to another @Message  / class of the reply message

Reply:
That was my first intention but it complicates a whole lot. The rpc even producer must be aware of the routing key used by the rpc consumer when sending a response. That can not be extracted in runtime from the `@message` annotation. If we want to have a reference to another message class for configuring the RPC call we'll need additional implementation, which i'm currently not sure how i'd do.

Reply:
I'll think this through again, as i agree with you change request.